frontend/js/src/settings/missing-data/MissingMBData.tsx

```diff
@@ -216,146 +221,216 @@ export default function MissingMBDataPage() {
-                    action={this.submitMissingData.bind(this, listen)}
-                  />
-                ); */
-
-                  const recordingMSID = getRecordingMSID(listen);
-                  const canDelete =
-                    Boolean(listen.listened_at) && Boolean(recordingMSID);
+                <FontAwesomeIcon icon={faLink} />
+              </button>
+            );
+            const listenCards = group.map((groupItem) => {
+              if (
+                deletedListens.find(
+                  (deletedMSID) => deletedMSID === groupItem.recording_msid
+                )
+              ) {
+                // If the item was deleted, don't show it to the user
+                return undefined;
+              }
+              let additionalActions;
+              const listen = missingDataToListen(groupItem, user);
+              const additionalMenuItems = [];
+              if (user?.auth_token) {
+                const recordingMSID = getRecordingMSID(listen);
+                const canDelete =
+                  Boolean(listen.listened_at) && Boolean(recordingMSID);
 
-                  if (canDelete) {
-                    additionalMenuItems.push(
-                      <ListenControl
-                        text="Delete Listen"
-                        icon={faTrashAlt}
-                        action={() => {
-                          deleteListen(data);
-                        }}
-                      />
-                    );
-                  }
+                if (canDelete) {
+                  additionalMenuItems.push(
+                    <ListenControl
+                      text="Delete Listen"
+                      icon={faTrashAlt}
+                      action={() => {
+                        deleteListen(groupItem);
+                      }}
+                    />
+                  );
+                }
 
-                  if (listen?.track_metadata?.additional_info?.recording_msid) {
-                    const linkWithMB = (
-                      <ListenControl
-                        buttonClassName="btn btn-sm btn-success"
-                        text=""
-                        title="Link with MusicBrainz"
-                        icon={faLink}
-                        action={() => {
-                          NiceModal.show(MBIDMappingModal, {
-                            listenToMap: listen,
+                if (listen?.track_metadata?.additional_info?.recording_msid) {
+                  const linkWithMB = (
+                    <ListenControl
+                      buttonClassName="btn btn-link color-orange"
+                      text=""
+                      title="Link with MusicBrainz"
+                      icon={faLink}
+                      action={() => {
+                        NiceModal.show<TrackMetadata, any>(MBIDMappingModal, {
+                          listenToMap: listen,
+                        }).then(({ recording_msid }) => {
```

Comment:
The `recording_msid` is undefined. Therefore the track is not removed from the `missingData` state.